CMakeLists.txt

```diff
@@ -365,14 +365,28 @@ if (PTHREAD_LIBRARY AND PTHREAD_INCLUDE_DIR)
 	message(STATUS "Pthread include dir: ${PTHREAD_INCLUDE_DIR}")
 elseif (WIN32 AND NOT MINGW)
 	# find pthread
-	find_path(PTHREAD_INCLUDE_DIR pthread.h HINTS C:/pthread-win32/include)
```

Comment:
Hmm. I just realized that this change will invalidate Windows building instructions we have in the README.md.

Is there a way to keep the old way working too or is it really one or the other?

Reply:
This is a good point; we certainly have to re-work the instructions. However, there is a choice what to do.

1. Simplify the instructions, and indicate to people that we have a new step to run 'nuget restore' to trigger download of the pthreads to the station (instead of the complex build step)
2. Extend some of CMakeLists to work the old way, and then extend the instructions to indicate how to use NuGet or manual build.

Following precedence set by referring to the OpenSSL download, I would encourage we choose 'option 1' and describe briefly how to use nuget to download and extract and reduce the complexity of the steps. We can leave a pointer on the instructions to the live build repository that is used to create the pthreads binary, and indicate people are welcome to fork and rebuild this as they wish (again, like we do for OpenSSL above).

Happy to change the readme.md as part of this PR if people are happy with me doing it 'option 1' style.

Reply:
OK, looking at the OpenSSL recommended link, the readme needs updating because they moved this to VS2017 builds... so we have an awesome blend of VS2013, 2015 and 2017 in the mix now. This PR will deprecate VS2013, but after this I will investigate a move of the SRT build to 2017 for SRT and PThreads NuGet, since then everything can be aligned on the same single VC runtime...

Reply:
Having seen what @maxlovic has done in a draft PR, it is clear option 1 is the right path. I don't want to duplicate his work though, and would happily just edit his work to be clearer with this once things are merged.

Marking resolved as 'option 1' being the choice - but not taking action since it would cause conflict with his efforts.

Reply:
`C:/pthread-win32/include` should stay.

Reply:
This is replaced by the more complex location GLOB steps that follow - within the nuget package, the 'includes' folder is inside the 'sources' folder now that pthreads is sourced as a precompiled binary rather than a complete project package.

When CMake runs, it prints out that when it locates a version if pthreads (since the folder name contains the build version, so we can't just have a static path) like this:
Pthread include dir: D:/Data/Dev/GIT/srt/packages/cinegy.pthreads-win64.2.9.1.15/sources

Reply:
This is more about the compliance to the current [README.md](https://github.com/Haivision/srt/blob/master/README.md#for-windows).
It tells to copy pthread files to "C:\pthread-win32\". Untill we change this, CMake has to look for pthreads in that path.

Reply:
I changed the CMake status if it does not find pthreads under the GLOB path to just an info, stating that it will rely instead on the already documented variables to pick up the path / lib.  I also added readme changes to this branch now reflecting a document way to grab with nuget - so you can take your pick of building and referencing the built elements by a VAR or using nuget and letting CMake locate it.

Reply:
This is more significantly changed, since it conflicted with the forewarned vcpkg changes. I hand-merged this, and it should now only add extra HINTS to the lookup and not break anything else... so hopefully this is permanently resolved now.